src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -2955,19 +2955,47 @@ Address MacroAssembler::add_memory_helper(const Address dst) {
   }
 }
 
-void MacroAssembler::add_memory_int64(const Address dst, int64_t imm) {
+void MacroAssembler::increment(const Address dst, int64_t value) {
+  assert(((dst.getMode() == Address::base_plus_offset &&
+           is_offset_in_range(dst.offset(), 12)) || is_imm_in_range(value, 12, 0)),
+          "invalid value and address mode combination");
   Address adr = add_memory_helper(dst);
-  assert_different_registers(adr.base(), t0);
+  assert(!adr.uses(t0), "invalid dst for address increment");
   ld(t0, adr);
-  addi(t0, t0, imm);
+  add(t0, t0, value, t1);
```

Comment:
It's not safe to clobber t1 sometimes. And I think it's better to limit the accepted value to 12 bits or less.

Reply:
In fact, we still need to trash t1 here when we have some unexpected memory address. Callers should be aware of this.